src/pke/lib/scheme/bgvrns/bgvrns-parametergeneration.cpp

```diff
@@ -198,8 +198,9 @@ std::pair<std::vector<NativeInteger>, uint32_t> ParameterGenerationBGVRNS::compu
             "Change parameters! Try reducing the number of additions per level, "
             "number of key switches per level, or the digit size. We cannot support moduli greater than 60 bits.");
     }
+
     uint32_t totalModSize = firstModSize;
-    moduliQ[0]            = FirstPrime<NativeInteger>(firstModSize, cyclOrder);
+    moduliQ[0]            = FirstPrime<NativeInteger>(firstModSize - 1, cyclOrder);
```

Comment:
I think this should be `LastPrime` with firstModSize

Reply:
a few unit tests would request LastPrime(19, 16384), which throws an exception because the returned prime is 18 bits. If LastPrime(19, 16384) is unrealistic, we can change the test parameters.

Reply:
We should increase the bit size in unit tests (by 1 or 2, whichever is enough)

Reply:
keeping FirstPrime<NativeInteger>(firstModSize - 1, cyclOrder);